scripts/analytics/premiumFeatureViewTime.js

```diff
@@ -49,49 +46,73 @@ co(function*() {
-    // user: { $regex: /^.{24}$/ }
-  }
-)).length;
-console.log("totalRecentUsers:", totalRecentUsers);
-
-const results = (yield analyticsDb.collection('log').mapReduce(function(){
-  emit('timeTotal', parseFloat(this.properties.timeViewed))
-  emit('usersThatViewedPremiumFeatures', 1)
-}, function(key, values){
-  return Array.sum(values);
-}, {
-  query: { event: "Premium Feature Viewed", _id: {$gt: startObjectId } },
-  out: {inline: 1}
-}));
-
-var usersThatViewedPremiumFeatures;
-var totalPremiumViewTime;
-
-results.forEach((obj)=>{
-  switch(obj._id){
-    case 'timeTotal':
-      totalPremiumViewTime = obj.value;
-    case 'usersThatViewedPremiumFeatures':
-      usersThatViewedPremiumFeatures = obj.value;
-  }
-})
-
-console.log("usersThatViewedPremiumFeatures:", usersThatViewedPremiumFeatures);
-console.log("totalPremiumViewTime:", totalPremiumViewTime, 'ms');
-console.log("Final G1KR2 value:", (totalPremiumViewTime/totalRecentUsers).toFixed(0), 'ms/user');
-
-mongoose.connection.close();
-analyticsDb.close();
+  var objectIdFromDate = function (date) {
+    return mongoose.Types.ObjectId(Math.floor(date.getTime() / 1000).toString(16) + "0000000000000000");
+  };
+
+  const startObjectId = objectIdFromDate(startDay);
+  const endObjectId = objectIdFromDate(endDay);
+
+  // First event was at 2016-01-19T12:54:05
+  // console.log("First premium view tracking event: ", (yield analyticsDb.collection('log').find({event: "Premium Feature Viewed"}).sort({_id:1}).limit(1).toArray())[0]._id);
+
+  console.log("Total premium view events:", yield analyticsDb.collection('log').count({ event: "Premium Feature Viewed", _id: { $gt: startObjectId, $lt: endObjectId} }));
+
+  const recentUserIds = (yield analyticsDb.collection('log').distinct('user',
+    {
+      _id: { $gt: startObjectId, $lt: endObjectId },
+      user: { $regex: /^.{24}$/ }
+    }
+  ));
+  
+  const totalRecentUsers = recentUserIds.length;
+  console.log("totalRecentUsers:", totalRecentUsers);
+  // console.log(JSON.stringify(recentUserIds.toString()).length);
+  
+  // const recentUsers = yield User.find({
+  //   _id: { $in: recentUserIds.map((strId)=>{ return mongoose.Types.ObjectId(strId) }) }
+  // })
+  //
+  // console.log("Recent users:", recentUsers.length);
+  
+  const recentIndividualUserIds = (yield User.find({
+    _id: { $in: recentUserIds.map((strId)=>{ return mongoose.Types.ObjectId(strId) }) },
+    role: { $nin: ['student', 'teacher', 'technology coordinator', 'advisor', 'principal', 'superintendent', 'parent'] }
+  }, {_id: true})).map((u)=>{return u._id + ''})
+  
+  const totalRecentIndividualUsers = recentIndividualUserIds.length;
+  
+  console.log("Recent Individual Accounts:", recentIndividualUserIds.length);
+  
+  
+  const results = (yield analyticsDb.collection('log').mapReduce(function(){
+    emit('timeTotal', parseFloat(this.properties.timeViewed));
+    emit('usersThatViewedPremiumFeatures', 1);
```

Comment:
Won't this repeatedly count users who viewed premium features multiple times? I guess it doesn't matter too much since it's not used in the KR value but then why is it measured?